### PR TITLE
run full demo as a test

### PR DESCRIPTION
**Pull request by v4hn**
Running that would have helped multiple times before.

Also simplify/streamline code in various places.

### CODE REVIEW

File: demo/test/pickplace_demo.test [hunk 1]
```diff
@@ -0,0 +1,15 @@
+<?xml version="1.0"?>
+<launch>
+    <include file="$(find moveit_resources_panda_moveit_config)/launch/demo.launch">
+        <arg name="use_rviz" value="false"/>
+        <!-- Enable for speedup once the argument is available upstream
+        <arg name="execution_type" value="last point" />
+        -->
```
Comment (v4hn) [inline demo/test/pickplace_demo.test:7]: Enabling this argument requires https://github.com/ros-planning/moveit_resources/pull/61 .
The speedup on my laptop is (surprisingly only) 18 -> 13 seconds.

**Comment by v4hn** [comment]
CI fails because the recent moveit_resources 0.7.2 release is not synced yet. Any idea when the next sync is scheduled?

**Comment by v4hn** [comment]
Do you want to fix your local system before getting this merged?
As CI succeeds and it works for me I would otherwise expect we can merge it now.
Thanks for the OBJECT lib cleanup. I use features ahead of time again...

**Comment by rhaschke** [comment]
Did you try to run the pick+place demo as well?

**Comment by v4hn** [comment]
```
❯ catkin run_tests --no-deps moveit_task_constructor_demo
[...]
❯ catkin_test_results build/moveit_task_constructor_demo/
Summary: 3 tests, 0 errors, 0 failures, 0 skipped
```

I believe this is what you mean? Locally, I run this on my Lunar-Linux machine though.

> Maybe add expectations on the number of found solutions

I prefer to leave out checks on the number of solutions for more complex tasks as they are also affected by compile options/computational resources/... The test checks for at least one solution which is sufficient for me.

**Comment by rhaschke** [comment]
> ```
> ❯ catkin run_tests --no-deps moveit_task_constructor_demo
> [...]
> ❯ catkin_test_results build/moveit_task_constructor_demo/
> Summary: 3 tests, 0 errors, 0 failures, 0 skipped
> ```
> 
> 
>     
>   
> 
> I believe this is what you mean? Locally, I run this on my Lunar-Linux machine though.

No. I meant (as stated in the [actual tutorial](https://ros-planning.github.io/moveit_tutorials/doc/moveit_task_constructor/moveit_task_constructor_tutorial.html) - sorry for the confusion):
```
roslaunch moveit_task_constructor_demo demo.launch
roslaunch moveit_task_constructor_demo pickplace.launch
```

**Review by rhaschke — APPROVED** [review]
Just tested again locally and today it works. No idea what was the problem the other day.

**Pull request merged**